### DISCUSSION TITLE
Query docs with overlapping date & time range

### DISCUSSION THREAD

**Discussion by chris-fellows** [ravendb/ravendb#15691] [Q&A]
RavenDB version: 5.4 (Dot NET Core)
My Project doc has properties StartDateTime (Date & Time) & EndDateTime (Date & Time). A project can last from days to 6 months. The time part of both properties is set. 

Is it possible to query all Project docs where at least some of the project was active within a filter date & time range? I do not just want projects that started and ended within the range. The filter date & time range could be anything from minutes to months.

**Comment by garayx** [comment]
Hi,

Isnt it simply checking if `StartDateTime` is between filter start date & end date ? can you post sample docs, and the expected query result?

**Reply by chris-fellows** [reply]
No. It needs to be logically implemented as checking if two date ranges fully OR partially overlap. I can do that in code but it's more efficient to do it in a query. For example, If there's a 1 month project starting 01/03/23 14:00 and a 6 month project starting on 01/01/23 14:00 then a filter 30/03/23 09:00 to 05/04/23 17:00 returns both projects because they were both active during the filter range.

**Reply by krivulcik** [reply]
Would something like this work well?

```
from index 'YourIndex'
where (ProjectStartTime < "2022-01-01T00:00:00.0000000" and ProjectEndTime > "2022-01-01T00:00:00.0000000")
or (ProjectStartTime < "2022-02-01T00:00:00.0000000" and ProjectEndTime > "2022-02-01T00:00:00.0000000")
```

Where 2022-01-01T00:00:00.0000000 is DateFrom and 2022-02-01T00:00:00.0000000 is DateTo.

In general:
```
from index 'YourIndex'
where (ProjectStartTime < FilterFrom and ProjectEndTime > FilterFrom)
or (ProjectStartTime < FilterTo and ProjectEndTime > FilterTo)
```
